client/my-sites/stats/hooks/use-is-advanced-feature-enabled.ts

```diff
@@ -0,0 +1,43 @@
+import config from '@automattic/calypso-config';
+import { useNoticeVisibilityQuery } from 'calypso/my-sites/stats/hooks/use-notice-visibility-query';
+import { useSelector } from 'calypso/state';
+import { canCurrentUser } from 'calypso/state/selectors/can-current-user';
+import { isJetpackSite } from 'calypso/state/sites/selectors';
+import useStatsPurchases from '../hooks/use-stats-purchases';
+
+const useIsAdvancedFeatureEnabled = ( siteId: number ) => {
+	const { isCommercialOwned, hasLoadedSitePurchases, isRequestingSitePurchases } =
+		useStatsPurchases( siteId );
+
+	const isSiteJetpackNotAtomic = useSelector( ( state ) =>
+		isJetpackSite( state, siteId, { treatAtomicAsJetpackSite: false } )
+	);
+
+	const canUserManageOptions = useSelector( ( state ) =>
+		canCurrentUser( state, siteId, 'manage_options' )
+	);
+	const canUserViewStats = useSelector( ( state ) =>
+		canCurrentUser( state, siteId, 'view_stats' )
+	);
+
+	const { isLoading: isLoadingNotices, data: purchaseNotPostponed } = useNoticeVisibilityQuery(
+		siteId,
+		'focus_jetpack_purchase',
+		canUserManageOptions
+	);
+
+	const isLoading = ! hasLoadedSitePurchases || isRequestingSitePurchases || isLoadingNotices;
+	const hasAdcancedPlan = isCommercialOwned;
+
+	// to redirect the user can't have a plan purached and can't have the flag true, if either is true the user either has a plan or is postponing
+	const advancedFeaturesEnabled =
+		config.isEnabled( 'stats/utm-module' ) &&
+		isSiteJetpackNotAtomic &&
+		hasAdcancedPlan &&
+		canUserViewStats && // this might be a redundant check
+		purchaseNotPostponed; // what about editors?
```

Comment:
@grzegorz-cp  Sorry I missed this - I think only personal users have the option to postpone the upgrade. Even if they did, they should be considered as free users, so none of the advanced features should be available to them. Based on those, I guess we could remove the condition here. What do you think?

Reply:
Right, it makes sense. We don't have to check for the postponed action if we check for commercial use. This will make the check simpler.